lib/iris/experimental/ugrid.py

```diff
@@ -737,358 +773,913 @@ def equal(self, other, lenient=None):
+                kwargs[normalise("edge", axis)] = coord
+        if face_coords_and_axes is not None:
+            for coord, axis in face_coords_and_axes:
+                kwargs[normalise("face", axis)] = coord
+
+        # check the UGRID minimum requirement for coordinates
+        if "node_x" not in kwargs:
+            emsg = (
+                "Require a node coordinate that is x-axis like to be provided."
+            )
+            raise ValueError(emsg)
+        if "node_y" not in kwargs:
+            emsg = (
+                "Require a node coordinate that is y-axis like to be provided."
+            )
+            raise ValueError(emsg)
+
+        if self.topology_dimension == 1:
+            self._coord_manager = _Mesh1DCoordinateManager(**kwargs)
+        elif self.topology_dimension == 2:
+            self._coord_manager = _Mesh2DCoordinateManager(**kwargs)
+        else:
+            emsg = f"Unsupported 'topology_dimension', got {topology_dimension!r}."
+            raise NotImplementedError(emsg)
+
+        # based on the topology_dimension, create the appropriate connectivity manager
+        # self._connectivity_manager = ...
+
+    def __eq__(self, other):
+        # TBD
+        return NotImplemented
+
+    def __getstate__(self):
+        # TBD
+        pass
+
+    def __ne__(self, other):
+        # TBD
+        return NotImplemented
+
+    def __repr__(self):
+        # TBD
+        args = []
+        return f"{self.__class__.__name__}({', '.join(args)})"
+
+    def __setstate__(self, state):
+        # TBD
+        pass
+
+    def __str__(self):
+        # TBD
+        args = []
+        return f"{self.__class__.__name__}({', '.join(args)})"
+
+    @property
+    def all_coords(self):
+        return self._coord_manager.all_members
+
+    @property
+    def edge_dimension(self):
+        return self._edge_dimension
+
+    @edge_dimension.setter
+    def edge_dimension(self, name):
+        if not name or not isinstance(name, str):
+            self._edge_dimension = f"Mesh{self.topology_dimension}d_edge"
+        else:
+            self._edge_dimension = name
+
+    @property
+    def edge_coords(self):
+        return self._coord_manager.edge_coords
+
+    @property
+    def face_dimension(self):
+        return self._face_dimension
+
+    @face_dimension.setter
+    def face_dimension(self, name):
+        if not name or not isinstance(name, str):
+            self._face_dimension = f"Mesh{self.topology_dimension}d_face"
+        else:
+            self._face_dimension = name
+
+    @property
+    def face_coords(self):
+        return self._coord_manager.face_coords
+
+    @property
+    def node_dimension(self):
+        return self._node_dimension
+
+    @node_dimension.setter
+    def node_dimension(self, name):
+        if not name or not isinstance(name, str):
+            self._node_dimension = f"Mesh{self.topology_dimension}d_node"
+        else:
+            self._node_dimension = name
```

Comment:
Should be stored in the `metadata_manager`

Reply:
Yeah, oops. Just fix it at your end 👍

Reply:
That's teach me to hack late at night 🤪